develop/debugging/pdb.rst

```diff
@@ -229,3 +238,93 @@ debug prompt.
+* https://pypi.python.org/pypi/debug
+
+pudb
+----
+It's an alternative to pdb with a curses interface. Its goal is to provide all the niceties of modern GUI-based debuggers in a more lightweight and keyboard-friendly package. PuDB allows you to debug code right where you write and test it–in a terminal. If you’ve worked with the excellent (but nowadays ancient) DOS-based Turbo Pascal or C tools, PuDB’s UI might look familiar.
+
+- Syntax-highlighted source, the stack, breakpoints and variables are all visible at once and continuously updated. This helps you be more aware of what’s going on in your program. Variable displays can be expanded, collapsed and have various customization options.
+- Simple, keyboard-based navigation using single keystrokes makes debugging quick and easy. PuDB understands cursor-keys and Vi shortcuts for navigation. Other keys are inspired by the corresponding pdb commands.
+- Use search to find relevant source code, or use “m” to invoke the module browser that shows loaded modules, lets you load new ones and reload existing ones.
+- Breakpoints can be set just by pointing at a source line and hitting “b” and then edited visually in the breakpoints window. Or hit “t” to run to the line under the cursor.
+- Drop to a Python shell in the current environment by pressing “!”.
+- PuDB places special emphasis on exception handling. A post-mortem mode makes it easy to retrace a crashing program’s last steps.
+- IPython integration (see wiki)
+
+* https://pypi.python.org/pypi/pudb
+
+q
+-
+
+Quick and dirty debugging output. All output goes to /tmp/q, which you can watch with this shell command: ``tail -f /tmp/q``. That way you can print variables, functions, etc. Check it's documentation for more examples.
+
+* https://pypi.python.org/pypi/q
+
+
+Debugging page templates
+========================
+
+Since Plone 5, Chameleon (five.pt) is used for the TAL engine. When using Chameleon, we can use the following snippet to debug page templates:
+
+Example::
+
+    <?python locals().update(econtext); import pdb; pdb.set_trace() ?>
+
+However, this doesn't work in skin templates and in TTW templates. If you want a full explanation of how this snippet works (specially about the econtext variable), check https://www.starzel.de/blog/magic-templates-in-plone-5.
+
+
+Debugging ZMI Python Script
+===========================
+
+If you install https://pypi.python.org/pypi/Products.enablesettrace in your instance, you can import pdb inside a Python Script.
+
+
+Browser Extensions
+==================
+
+If you need to call ``/@@reload`` (if you installed plone.reload) or ``?diazo`` on your current Plone, you can use the Plone Reloader extension.
+
+* https://chrome.google.com/webstore/detail/plone-reloader/bcdahfmmenfikninekekpbncgdkdlapl
+
+This extension displays the plone.reload form in a popup so you can easily reload your current Plone instance code without switching to another tab. It also provides buttons to open diazo off/debug urls.
```

Comment:
Same here, here is some work on reST needed and some changes to follow our style-guide.

:)